anaconda_anon_usage/tokens.py

```diff
@@ -117,6 +122,31 @@ def environment_token(prefix=None):
     return _saved_token(fpath, "environment", prefix)
 
 
+@cached
+def anaconda_token():
+    """
+    Returns the token for the logged-in anaconda user, if present.
+    """
+    try:
+        from anaconda_cloud_auth.config import AnacondaCloudConfig
```

Comment:
Just want to add for the record that this puts Pydantic into the call path for every conda installation that has anaconda-cloud-auth installed (all Anaconda Distribution eventually?). Is that the only way to load the API key or is there a lower-level API that could work around the config loading?

Reply:
I've considered three somewhat separate options:

- In this case, if the import fails, the token will not be included. So we're making anaconda-cloud-auth and its downstream dependencies are optional. If the user has installed things like Anaconda Assistant, conda scan, etc., then good news—those dependencies will be there, _and_ the user will be logged in.

- I investigated directly loading `~/.anaconda/keyring` using the standard library. It works fine! We can add that as a fallback. But this won't work if we move to direct `keychain` support.

- In theory, we could also use `keychain` as an optional dependency too, and use the system-stored keychain.

We can try any combination of the three of these! And I am happy to noodle on this until we get the best combination. What do you think we should consider?

Reply:
OK, I've chosen to use `keyring` and standard file operations to read the token from the system and file locations, respectively. I'm taking the one with the latest expiration date if both are present